pkg/build/build.go

```diff
@@ -111,17 +172,67 @@ type Options struct {
 	ValidateRemoteImageDigests bool
 }
 
+const (
+	DefaultBucket                     = release.CIBucketK8sInfra
+	DefaultBuildDir                   = ""
+	DefaultBuildType                  = "ci"
+	DefaultRegistryName               = release.GCRIOPathCI
+	DefaultGCSRoot                    = DefaultBuildType
+	DefaultVersion                    = ""
+	DefaultVersionSuffix              = ""
+	DefaultCI                         = true
+	DefaultConfigureDocker            = false
+	DefaultFast                       = false
+	DefaultAllowDup                   = true
+	DefaultNoUpdateLatest             = false
+	DefaultPrivateBucket              = false
+	DefaultValidateRemoteImageDigests = false
+)
+
+var (
+	DefaultExtraVersionMarkers = []string{}
+
+	DefaultOptions = &Options{
+		Fs:    afero.NewOsFs(),
+		Store: object.NewDefaultGCS(),
```

Comment:
We could consider moving these out of `Options` as just fields in `DefaultClient` and introducing a `ClientOptFn(d *DefaultClient)` type that can be passed as a variadic arg to `New(opts ...ClientOptFn)`. Then have four implementations:
- `WithFs(d *DefaultClient)`
- `WithStore(d *DefaultClient)`
- `WithRegistry(d *DefaultClient)`
- `WithOptions(d *DefaultClient)` -- we could consider just making all of these fields also instead of a substruct

Then in the `New()` body we build the default client then run the options on it before returning:
```
for _, f := range opts {
  f(r)
}
```
To override with any values. This allows us to have a single `New()` rather than `New` and `NewDefault`.